dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureDefinitionUtils.java

```diff
@@ -391,6 +399,38 @@ static int getIntegerValue(String value, int defaultValue) {
             manifest = loadManifest(mfFile, inStream);
         }
 
+        /**
+         * Get alternate names of the feature if any.
+         *
+         * @return A (possibly empty) set of alternate names.
+         */
+        Set<String> getAltNames() {
+
+            if (alternateNames == null) {
+                Set<String> result;
+                String ibmAltNames;
+                try {
+                    ibmAltNames = getMainAttributeValue(IBM_ALT_NAMES);
+                } catch (IOException e) {
+                    return Collections.<String> emptySet();
+                }
+
+                if (ibmAltNames == null) {
+                    result = Collections.<String> emptySet();
+                } else {
+                    Map<String, Map<String, String>> data = ManifestHeaderProcessor.parseImportString(ibmAltNames);
+
+                    result = new HashSet<String>();
+                    for (Map.Entry<String, Map<String, String>> entry : data.entrySet()) {
+                        //any attributes are not supported and will be ignored.
+                        result.add(entry.getKey());
+                    }
```

Comment:
This could be replaced by this line:
`result = new ArrayList<String>(data.keySet());`
That is assuming you change the type to a Collection<String> otherwise it would be:
`result = new HashSet<String>(data.keySet());`

Reply:
suggestion adopted. Thanks